public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/SimplifiedRuleEditor.test.tsx

```diff
@@ -137,27 +136,70 @@ describe('Can create a new grafana managed alert using simplified routing', () =
-      await user.click(byRole('combobox').get(contactPointInput));
-      await clickSelectOption(contactPointInput, 'lotsa-emails');
+      await selectContactPoint(user, 'Email');
 
       expect(await screen.findByText('Email')).toBeInTheDocument();
     });
   });
-});
+  describe('switch modes enabled', () => {
+    testWithFeatureToggles(['alertingQueryAndExpressionsStepMode', 'alertingNotificationsStepMode']);
+
+    it('can create the new grafana-managed rule with default modes', async () => {
+      const contactPointName = 'lotsa-emails';
+      const capture = captureRequests((r) => r.method === 'POST' && r.url.includes('/api/ruler/'));
+
+      const { user } = renderRuleEditor();
+
+      await user.type(await ui.inputs.name.find(), 'my great new rule');
+
+      await selectFolderAndGroup(user);
+
+      await selectContactPoint(user, contactPointName);
 
-function renderSimplifiedRuleEditor() {
-  return render(
-    <AlertmanagerProvider alertmanagerSourceName={GRAFANA_DATASOURCE_NAME} accessType="notification">
-      <Routes>
-        <Route path={'/alerting/new/:type'} element={<RuleEditor />} />
-        <Route path={'/alerting/:id/edit'} element={<RuleEditor />} />
-      </Routes>
-    </AlertmanagerProvider>,
-    { historyOptions: { initialEntries: ['/alerting/new/alerting'] } }
-  );
-}
+      // save and check what was sent to backend
+      await user.click(ui.buttons.saveAndExit.get());
+      const requests = await capture;
+      const serializedRequests = await serializeRequests(requests);
+      expect(serializedRequests).toMatchSnapshot();
+    });
+    it('can create the new grafana-managed rule with advanced modes', async () => {
+      const capture = captureRequests((r) => r.method === 'POST' && r.url.includes('/api/ruler/'));
+
+      const { user } = renderRuleEditor();
+
+      await user.click(ui.inputs.switchModeBasic(2).get()); // switch to query step advanced mode
+      await user.click(ui.inputs.switchModeBasic(5).get()); // switch to notifications step advanced mode
+      await user.type(await ui.inputs.name.find(), 'my great new rule');
+
+      await selectFolderAndGroup(user);
+
+      // save and check what was sent to backend
+      await user.click(ui.buttons.saveAndExit.get());
+      const requests = await capture;
+      const serializedRequests = await serializeRequests(requests);
+      expect(serializedRequests).toMatchSnapshot();
+    });
+    it('can create the new grafana-managed rule with only notifications step advanced mode', async () => {
+      const capture = captureRequests((r) => r.method === 'POST' && r.url.includes('/api/ruler/'));
+
+      const { user } = renderRuleEditor();
+
+      await user.type(await ui.inputs.name.find(), 'my great new rule');
+
+      await selectFolderAndGroup(user);
+
+      // // from the previous test, we are in advanced mode for both query and notifications steps
```

Comment:
I'd avoid relying on previous tests. Is there any way to ensure we see correct mode in this tests? Maybe by mocking localStorage and setting proper value there?

Reply:
Actually the idea is testing the real functionality that causes this behaviour. We want to test this local storage logic.

Reply:
ok, I understand what you mean. I think it makes sense.

Reply:
done